src/core/reducers/api.js

```diff
@@ -7,8 +8,44 @@ import {
   SET_CLIENT_APP,
   SET_USER_AGENT,
 } from 'core/constants';
+import type {
+  SetAuthTokenAction,
+  LogOutUserAction,
+  SetClientAppAction,
+  SetLangAction,
+  SetUserAgentAction,
+} from 'core/actions/index';
+import type { Exact } from 'core/types/coreTypes';
 
-export default function api(state = {}, action) {
+type UserAgentInfoType = {|
+  browser: string,
+  os: string,
+|};
+
+export type ApiStateType = {
+  token: ?string,
+  lang: ?string,
+  clientApp: ?string,
+  userAgent: ?string,
+  userAgentInfo: ?UserAgentInfoType,
+};
+
+export const initialApiState = {
+  token: null,
+  lang: null,
+  clientApp: null,
+  userAgent: null,
+  userAgentInfo: null,
+};
+
+export default function api(
```

Comment:
Some of this syntax is a bit hairy to someone new to flow. Any chance there's something you could point to (a docs page on what this is) for more foreign syntax like this? I don't even know what to search for.